### DISCUSSION TITLE
Display changes in tile function from v1 to current

### DISCUSSION THREAD

**Discussion by mike-amodeo** [cogeotiff/rio-tiler#495] [Q&A] [answered]
I have a map tile service in production that was built in spring of 2020 off an old version of rio-tiler. I am not sure of the exact version, but it was version 1 because we are calling the `tile` function from `main`, which was renamed in version 2.0a1. We are preparing an update and were going to use the current version. However, we are getting different visual results and we cannot replicate the older output. 

In the old version, we called `tile` from main without any options, and get this smoothed output
![image](https://user-images.githubusercontent.com/21199337/163824992-bc7f0efd-c01c-4f02-95a6-a63fcc657ae1.png)

In the new version, we are calling `tile` from `COGReader` within a call of `mosaic_reader` 
```
def reader(file, x, y, z, **kwargs):
        with COGReader(file) as image:
            return image.tile(x, y, z, **kwargs)

    try:
        img, _ = mosaic_reader(
            full_paths,
            reader,
            int(request["x"]),
            int(request["y"]),
            int(request["z"]),
            allowed_exceptions=(TileOutsideBounds, RasterioIOError),
        )
        tile = img.data[0]
        mask = img.mask
```
and getting results like below.
![image](https://user-images.githubusercontent.com/21199337/163825072-a042c095-ae63-4a7b-88dc-38952f68abb7.png)

We want to replicate the smoothing along the color steps and have tried a few options within `COGReader` for the `resampling_methods` but have not had success yet. Was there a default setting in the old version that got changed in version 2 of rio-tiler that would cause this change? Is there a way for us to replicate the old experience in the new version with different settings? Thank you!

**Comment by vincentsarago** [comment]
@mike-amodeo sorry about this. 

in ~1.x the `tile` method was using `resampling_methods="bilinear"` https://github.com/cogeotiff/rio-tiler/blob/4286e55d43172040b4027575a845532e55a0fdf9/rio_tiler/utils.py#L343-L344 but in the current version we use `nearest` (which is a safest choice) https://github.com/cogeotiff/rio-tiler/blob/f72d57c319675e3431a059abe7c34d96ce86f848/rio_tiler/reader.py#L33

**Comment by mike-amodeo** [comment]
Thanks, @vincentsarago . Not a problem. But when we tried adding the `resampling_methods=bilinear` in the call of `COGReader`, it was not working. Perhaps we had the wrong syntax. What is the best way to call this given our setup? Does this need to be in the `mosaic_reader` function or in `tile`?

**Comment by vincentsarago** [comment, marked as answer]
```python
def reader(file, x, y, z, **kwargs):
        with COGReader(file) as image:
            return image.tile(x, y, z, **kwargs)

    try:
        img, _ = mosaic_reader(
            full_paths,
            reader,
            int(request["x"]),
            int(request["y"]),
            int(request["z"]),
            allowed_exceptions=(TileOutsideBounds, RasterioIOError),
            resampling_method="bilinear",
        )
        tile = img.data[0]
        mask = img.mask
```

☝️ should give the same result 🤞
